go/interpreter/lfvm/converter.go

```diff
@@ -47,7 +48,8 @@ func NewConverter(config ConversionConfig) (*Converter, error) {
 	var cache *lru.Cache[tosca.Hash, Code]
 	if config.CacheSize > 0 {
 		var err error
-		capacity := config.CacheSize / maxCachedCodeLength
+		const instructionSize = int(unsafe.Sizeof(Instruction{}))
+		capacity := config.CacheSize / maxCachedCodeLength / instructionSize
```

Comment:
The LRU cache we are using can only be configured to a maximum number of elements (=capacity). However, the elements we are storing in it are pairs of hashes and codes. Hashes have a fixed size of 32 byte, but codes can be up to ~24KB large.

We want to give the user the option to set an upper boundary on the used memory in bytes. This should prevent the need for the user to figure out the conversion factor between cache capacity in elements and actual memory usage.

 This code here computes the maximum capacity for the cache given the user defined memory limit, assuming the worst-case code size and ignoring the overhead cost for the LRU map and the hash keys. 

So technically it might be possible for the cache to be a few percent larger than the given memory budget (due to the keys and the LRU overhead), but practically it is unlikely that all codes have the maximum length. Most are way shorter.

The important thing, however, is that an upper limit is enforced that is close to what the user has defined.

Reply:
yes, that makes sense now.